platform-sdk/docs/proposals/DP-0x/0X - Tss block signing - Cryptography APIs.md

```diff
@@ -0,0 +1,498 @@
+P₄  2        P₄_EC_PublicKey
+P₅  1        P₅_EC_PublicKey
+P₆  1        P₆_EC_PublicKey
+```
+A threshold:
+`5`
+
+An old ownership map: `ShareId`->`Participant`:
+```
+sid₁	sid₂	sid₃	sid₄	sid₅	sid₆	sid₇	sid₈	sid₉	sid₁₀
+P₁  	P₁  	P₁  	P₁  	P₁  	P₂  	P₂  	P₃  	P₄  	P₄  
+```
+
+And a new ownership map: `ShareId`->`Participant`:
+```
+sid₁	sid₂	sid₃	sid₄	sid₅	sid₆	sid₇	sid₈	sid₉	sid₁₀
+P₁  	P₁  	P₁  	P₂  	P₂  	P₃  	P₄  	P₄  	P₅  	P₆  
+```
+
+but instead, each participant will generate a `TssMessage` out of each `ProprietaryShare` owned by the participant.
+Once finished, the list of ProprietariesShares will be updated 
+
+[TODO]
+
+
+### Architecture
+
+![img_6.png](img_6.png)
+
+1. **TSS Library**: Consensus node will use the TSS library to create shares, create TSS messages to send to other nodes, assemble shared public keys (ledgerId), and sign the block-node merkle tree hash.
+2. **Bilinear Pairings Signature Library**: Provides cryptographic objects (PrivateKey, PublicKey, and Signature) and operations for the block-node and consensus-node to sign and verify the signatures in the block. Consensus-node uses this library indirectly through the TSS Library.
+3. **Bilinear Pairings API**: Generalization to be included at compilation time providing the cryptography primitives and the arithmetic operations for working with a specific EC curve and the underlying Groups, Fields, and Pairings.
+4. **Bilinear Pairings Implementation**: Java underlying implementation of the previous API, that will be loaded at runtime using Java's SPI. Multiple implementations can be provided for supporting different types of curves.
+5. **Native Support Library**: Provides a set of generic functions loading native libraries in different system architectures when packaged in a jar using a predefined organization, so they can be accessed with JNI.
+6. **Arkworks Rust Library**: arkworks is a Rust ecosystem for zkSNARK programming. Libraries in the arkworks ecosystem provide efficient implementations of all components required to implement zkSNARK applications, from generic finite fields to R1CS constraints for common functionalities.
+7. **EC-Key Utils**: Utility module to enable the node operator to generate pre-genesis initial public/private Key pair
+
+### Module Organization and repositories
+![img_7.png](img_7.png)
+1. **hedera-cryptography**: This is a separate repository for hosting cryptography-related libraries. This repository is necessary as a means of facilitating our build process that includes Rust libraries. Also provides independent release cycles between consensus node code and block node code.
+2. **swirlds-native-support**: Gradle module that enables loading into memory compiled native libraries, so they can be used with JNI.
+3. **swirlds-cryptography-tss**: Gradle module for the TSS Library. This library-only client is the consensus node, so will remain close to it in `hedera-services` repository under `platform-sdk` folder.
```

Comment:
```suggestion
3. **swirlds-cryptography-tss**: Gradle module for the TSS Library. This library's only client is the consensus node, so it will remain close to it in `hedera-services` repository under `platform-sdk` folder.
```